Yes I know that patch is suboptimal. It should do for now ;)

--- perl-5.8.0/ext/Errno/Errno_pm.PL.orig	2003-09-20 18:41:01.000000000 +0200
+++ perl-5.8.0/ext/Errno/Errno_pm.PL	2003-09-20 18:41:31.000000000 +0200
@@ -109,7 +109,7 @@
 	     ) {
 	# Some Linuxes have weird errno.hs which generate
 	# no #file or #line directives
-	$file{'/usr/include/errno.h'} = 1;
+	$file{'/usr/dietlibc/include/errno.h'} = 1;
     } elsif ($^O eq 'MacOS') {
 	# note that we are only getting the GUSI errno's here ...
 	# we might miss out on compiler-specific ones
